packages/next/src/server/app-render/app-render.tsx

```diff
@@ -522,12 +522,29 @@ async function generateDynamicFlightRenderResult(
     onFlightDataRenderError
   )
 
-  const rscPayload = await generateDynamicRSCPayload(ctx, options)
+  const RSCPayload = await generateDynamicRSCPayload(ctx, options)
+
+  if (
+    process.env.NODE_ENV === 'development' &&
+    renderOpts.experimental.dynamicIO
+  ) {
+    const [resolveValidation, validationOutlet] = createValidationOutlet()
+    ;(RSCPayload as any)._validation = validationOutlet
```

Comment:
we should probably add this as an optional property on the type and just indicate it's for dev only